### ISSUE TITLE
Unable to pay invoice with docker deployment

### ISSUE THREAD

**Issue by sidaudhi** [ElementsProject/lightning#2377]
### Issue and Steps to Reproduce
The issue seems to be related to #2282 but specific to the docker image elementsproject/lightningd
Using docker-compose i re-installed the container after pulling the latest image and when I execute:
```
docker exec lightning-cli pay <bolt1>
```
I get the error:
```
{ "code" : -32601, "message" : "Unknown command 'pay'" }
```
### `getinfo` output
```
{
  "id": "0230d70b0506aa447a1db407cc04e41448ab92bbd373f5c9a1029a40f357fcdcad",
  "alias": "LN.pay",
  "color": "20b0cf",
  "num_peers": 11,
  "num_pending_channels": 2,
  "num_active_channels": 9,
  "num_inactive_channels": 0,
  "address": [
    {
      "type": "ipv4",
      "address": "138.68.231.196",
      "port": 9735
    }
  ],
  "binding": [
    {
      "type": "ipv4",
      "address": "0.0.0.0",
      "port": 9735
    }
  ],
  "version": "v0.6.3rc1-118-g0da4054",
  "blockheight": 564007,
  "network": "bitcoin",
  "msatoshi_fees_collected": 0
}
```

**Comment by wailo** [comment]
I am experiencing the same. Using both the provided docker image and building the sourcecode manually.

**Comment by wailo** [comment]
Managed to get it work. Had to pass `--plugin-dir=PATH-TO-PAY-PLUGIN` to lightningd process

**Comment by sidaudhi** [comment]
How would this look like in a docker deployment? I tried this:

```
clightning_bitcoin:
    image: elementsproject/lightningd
    container_name: lightning
    command:
      - --bitcoin-rpcconnect=bitcoind
      - --bitcoin-rpcuser=rpcuser
      - --bitcoin-rpcpassword=rpcpass
      - --network=bitcoin
      - --alias=LN.pay
      - --rgb=20B0CF
      - --log-level=debug
      - --bind-addr=0.0.0.0:9735
      - --announce-addr=138.68.231.196:9735
      - --fee-base=0
      - --fee-per-satoshi=6000
      - --plugin-dir=/root/plugins
    environment:
      EXPOSE_TCP: "true"
    expose:
      - "9735"
    ports:
      - target: 9735
        published: 9735
        protocol: tcp
        mode: host
    volumes:
      - type: bind
        source: "/mnt/volume_sfo2_01/bitcoin/mainnet/clightning"
        target: "/root/.lightning"
      - type: bind
        source: "/mnt/volume_sfo2_01/plugins"
        target: "/root/plugins"
      - type: bind
        source: "/mnt/volume_sfo2_01/bitcoin/mainnet/bitcoind"
        target: "/etc/bitcoin"
```
I manually moved the plugins folder from source into /mnt/volume_sfo2_01. 

Should this work?

**Comment by wailo** [comment]
Dockerfile need to updated as well. I have created a PR https://github.com/ElementsProject/lightning/pull/2418 to fix this issue.

Until the fix is merged, you can fix it by following these steps

1. Remove existing images `docker-compose down`
2. Clone the repo
3. Import the changes in Dockerfile from the PR.
4. Import the changes in  docker-compose.yaml from the PR, and instruct docker-compose to build clightning_bitcoin from the dockerfile instead of pulling the image from the server. I paste the content below
5. The directory structure should look like this:
 ├── docker-compose.yaml
 ├── lightning
6. Where lightning folder is the cloned repo
7. Run docker-compose build to rebuild the images.

```
version: "3"
services:
  bitcoind:
    image: nicolasdorier/docker-bitcoin:0.16.3
    container_name: bitcoind
    environment:
      BITCOIN_EXTRA_ARGS: |
        testnet=1
        whitelist=0.0.0.0/0
        server=1
        rpcuser=rpcuser
        rpcpassword=rpcpass
    expose:
      - "18332"
    ports:
      - "0.0.0.0:18333:18333"
    volumes:
      - "bitcoin_datadir:/data"

  clightning_bitcoin:
    # image: elementsproject/lightningd
    build: lightning
    container_name: lightningd
    command:
      - --bitcoin-rpcconnect=bitcoind
      - --bitcoin-rpcuser=rpcuser
      - --bitcoin-rpcpassword=rpcpass
      - --plugin-dir=/usr/libexec/c-lightning/plugins
      - --network=testnet
      - --alias=myawesomenode
      - --log-level=debug
    environment:
      EXPOSE_TCP: "true"
    expose:
      - "9735"
    ports:
      - "0.0.0.0:9735:9735"
    volumes:
      - "clightning_bitcoin_datadir:/root/.lightning"
      - "bitcoin_datadir:/etc/bitcoin"
    links:
      - bitcoind

volumes:
  bitcoin_datadir:
  clightning_bitcoin_datadir:

```

**Comment by max302** [comment]
Considering the addition of plugins as an official feature in 0.7, it might be smart to add a volume dedicated to plugins like @sidaudhi did by default. 

I have a similar setup and everything works just fine now.

**Comment by cdecker** [comment]
Fixed by #2451
